src/createEffectCapableStore.js

```diff
@@ -40,9 +40,21 @@ export const wrapDispatch = store => action => {
     `It's allowed to yield only functions (side effect)`);
 
   // Effects are executed after action is dispatched
-  effects.forEach(effect => effect(wrapDispatch(store)));
+  const effectPromises = effects.map(effect => {
+    const effectResult = effect(wrapDispatch(store));
 
-  return result;
+    // TODO: this should obviously be something like isThenable
+    if (!(effectResult instanceof Promise)) {
+      return new Promise(resolve => resolve(effectResult));
+    } else {
+      return effectResult;
+    }
+  });
+
+  return {
+    action: result,
+    effectPromises
```

Comment:
You might not have realized that but the wrapped `dispatch` function emulates middlewares, we could achieve the same using middlewares but I wanted to avoid that (so that user don't need to use store enhancer + middleware).

Also, look at the Redux [docs](http://redux.js.org/docs/advanced/Middleware.html#seven-examples) it's pretty common to modify returned action, in fact most middlewares are doing that.

It won't break any API with existing code it only depends how you compose your `applyMiddleware` and `storeFactory`.